src/lib/datatypes/sol-str-table.c

```diff
@@ -30,16 +30,21 @@ sol_str_table_lookup_fallback(const struct sol_str_table *table,
     const struct sol_str_table *iter;
     uint16_t len;
 
-    if (SOL_UNLIKELY(key.len > INT16_MAX))
+    if (SOL_UNLIKELY(key.len > INT16_MAX)) {
+        errno = EINVAL;
         return fallback;
+    }
 
     len = (uint16_t)key.len;
 
     for (iter = table; iter->key; iter++) {
         if (iter->len == len && memcmp(iter->key, key.data, len) == 0) {
+            errno = 0;
             return iter->val;
         }
     }
+
+    errno = ENOENT;
     return fallback;
```

Comment:
The doc does not say anything about set `errno` to `EINVAL` i guess would be good tells that in case the error `errno` will be set properly.